jobs/client.go

```diff
@@ -0,0 +1,80 @@
+package jobs
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	urlUtils "net/url"
+	"os"
+	"strconv"
+)
+
+const (
+	parserError        = "could not parse error"
+	unauthorizedAccess = "unauthorized access"
+	urlParseError      = "could not parse base url"
+)
+
+type ZipClient struct {
+	BaseUrl string
+	ApiKey  string
+}
+
+func NewZipClient() Client {
+	apiKey := os.Getenv("ZIP_RECRUITER_KEY")
+	if len(apiKey) == 0 {
+		panic("ZIP_RECRUITER_KEY env variable not set")
+	}
+	baseUrl := os.Getenv("ZIP_RECRUITER_BASE_URL")
+	if len(baseUrl) == 0 {
+		panic("ZIP_RECRUITER_BASE_URL env variable not set")
+	}
+	return &ZipClient{BaseUrl: baseUrl, ApiKey: apiKey}
+}
+
+func (z *ZipClient) Get(request ZipRequest) (*ZipResponse, error) {
+	u, err := urlUtils.Parse(z.BaseUrl)
+	if err != nil {
+		log.Printf("Error: %s", urlParseError)
+		return nil, errors.New(urlParseError)
+	}
+	queryParams := u.Query()
+	queryParams.Set("search", request.Search)
+	queryParams.Set("location", request.Location)
+	queryParams.Set("radius_miles", strconv.Itoa(int(request.RadiusMiles)))
+	queryParams.Set("days_ago", strconv.Itoa(int(request.DaysAgo)))
+	queryParams.Set("jobs_per_page", strconv.Itoa(int(request.JobsPerPage)))
+	queryParams.Set("page", strconv.Itoa(int(request.Page)))
+	queryParams.Set("refine_by_salary", strconv.Itoa(int(request.RefineSalary)))
+	queryParams.Set("api_key", z.ApiKey)
+	u.RawQuery = queryParams.Encode()
```

Comment:
this method seems large, please move this code to other private method

Reply:
done

Reply:
done